sw/device/lib/dif/dif_kmac.c

```diff
@@ -311,13 +311,22 @@ dif_kmac_result_t dif_kmac_absorb(dif_kmac_t *kmac, const void *msg, size_t len,
     return kDifKmacError;
   }
 
-  // Copy the message one byte at a time.
-  // This could be sped up copying a word at a time but be careful
-  // about message endianness (e.g. only copy a word at a time when in
-  // little-endian mode).
-  for (size_t i = 0; i < len; ++i) {
+  // Copy message using aligned word sized loads and stores where possible to
+  // improve performance. Note: the parts of the message copied a byte at a time
+  // will not be byte swapped in big-endian mode.
+  const uint8_t *data = (const uint8_t *)msg;
+  for (; len != 0 && ((uintptr_t)data) % sizeof(uint32_t); --len) {
```

Comment:
`((uintptr_t)data) % sizeof(uint32_t)` is very expensive. Why not `(uintptr_t)data & 3`?

Reply:
The number of times that I've written `foo & 2` or `foo % 1` and had to debug why my code is doing something stupid makes me prefer just using `%` all the time and relying on the compiler to sort it out! :-)

Reply:
Thanks for the clarification @mundaym and @rswarbrick. It is ok to leave as is.